assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -402,8 +517,58 @@ export const controls = {
 	),
 };
 
+// eslint-disable-next-line complexity
 export const reducer = createReducer( ( state, { type, payload } ) => {
 	switch ( type ) {
+		case INSERT_NOTIFICATION_INTO_RESOLVED_QUEUE: {
+			const { id } = payload;
+
+			/**
+			 * The notification we want to add to the already-resolved queue.
+			 */
+			const notification = state.notifications?.[ id ];
+
+			if ( notification === undefined ) {
+				global.console.warn(
+					`Could not add notification with ID "${ id }" to queue. Notification "${ id }" is not registered.`
+				);
+
+				break;
+			}
+
+			state.queuedNotifications[ notification.groupID ] =
+				state.queuedNotifications[ notification.groupID ] || [];
```

Comment:
This is a side-effect which could make the queue appear to be resolved if it wasn't before. If `queuedNotifications` isn't an array yet, then the queue hasn't been resolved (or could be in process but not enqueued any yet). In this scenario I think we just return early, correct? I think we simply check if it's an array and if not return rather than setting it here.

Reply:
Oh that's better, sounds good 👍🏻